### PR TITLE
Associate project for every organisation fetched

### PR DESCRIPTION
**Pull request by YashviJain01**
### :nut_and_bolt: Description: What code changed, and why?
When a organisation is migrated from the chef server to automate, an IAM project needs to be linked with it. 
So added the call to project create service to create the project with the name of serverId_orgID, every time the organisation is stored in the database the project is created with the name having all the policies as editors, viewer, owner.

### :+1: Definition of Done
Project is created and stored in database at the time of migration

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs

Request
This is a get request.
```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/infraserverorgs'```

Response

```
{
    "orgs": [
        {
            "id": "test",
            "name": "test",
            "admin_user": "",
            "credential_id": "",
            "server_id": "serverId",
            "projects": ["serverID_test"]
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Comment by m0dul3** [comment]
a

File: components/infra-proxy-service/cmd/infra-proxy-service/commands/serve.go [hunk 1]
```diff
@@ -97,7 +99,8 @@ func serve(cmd *cobra.Command, args []string) {
 	// get secrets client
 	secretsClient := secrets.NewSecretsServiceClient(secretsConn)
 
-	service, err := service.Start(l, migrationConfig, connFactory, secretsClient, authzClient)
+	service, err := service.Start(l, migrationConfig, connFactory, secretsClient, authzClient, authzClientProject)
```
Comment (kalroy) [inline components/infra-proxy-service/cmd/infra-proxy-service/commands/serve.go:102]: `authzProjectClient` can be a better name of the variable?
```suggestion
	service, err := service.Start(l, migrationConfig, connFactory, secretsClient, authzClient, authzProjectClient)
```

File: components/infra-proxy-service/server/orgs.go [hunk 2]
```diff
@@ -240,7 +241,11 @@ func (s *Server) GetInfraServerOrgs(ctx context.Context, req *request.GetInfraSe
 	// Save organisations in backend DB
 	orgs := []storage.Org{}
 	for key := range orgsList {
-		org, err := s.service.Storage.StoreOrg(ctx, key, key, "", "", req.ServerId, nil)
+		projects, err := createProjectFromOrgIdAndServerID(s, ctx, req.ServerId, key)
```
Comment (kalroy) [inline components/infra-proxy-service/server/orgs.go:244]: Will it create the policies too for the project?

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/infra-proxy-service/server/orgs.go (hunk 2 above)
Comment (YashviJain01) [reply components/infra-proxy-service/server/orgs.go:244]: Yes, we are passing skip policies as false, therefore its creating project with all the policies.

File: components/infra-proxy-service/cmd/infra-proxy-service/commands/serve.go (hunk 1 above)
Comment (YashviJain01) [reply components/infra-proxy-service/cmd/infra-proxy-service/commands/serve.go]: Changed to the suggested name

**Pull request closed without merging**